### DISCUSSION TITLE
iconv not working the way it should

### DISCUSSION THREAD

**Discussion by lbdyck** [zopencommunity/meta#745] [General]
I tried the following command using the `zot` version of `iconv` and it fails big time as it does not understand `-F`

`iconv -F -t IBM-1047 envdelta.sh`

Then I tried:

`iconv -F -t IBM-1047 envdelta.sh`

And it failed with the converted data being garbage.

Then I tried 

`/bin/iconv -F -t IBM-1047 envdelta.sh`

and it worked perfectly.

But the `man` page on the system is the one from z/OS and not from `zot` assuming there is an updated `man` page.

**Comment by IgorTodorovskiIBM** [comment]
For context:
```
  -F
           Uses the input file's coded character set (as defined in the file tag)
           as the source code set. If -f is also specified, and the oldset matches
           the file tag or if no file tag code is set, then oldset is used as the
           source code set. If -F and -f are specified and oldset does not match
           the file tag code set, then iconv fails with an error.
```
We could employ the same technique as cat and cp where if we detect an invalid arg, we call the /bin version.

**Comment by lbdyck** [comment]
or add the z/OS function to the open-source and upstream it so there is only one tool

**Reply by IgorTodorovskiIBM** [reply]
Yeah, that's probably the better approach here. The offloading to /bin/cp in zopen cp was mainly for the dataset support which will take  considerably more effort than this.

**Comment by lbdyck** [comment]
tis a shame the z/OS source can't be open-sourced so you could take it and upstream the z/OS distinctives :)

**Comment by lbdyck** [comment]
Had issue with `zigi` where the `iconv` of .gitattributes was resulting in a file that was not readable - changed `iconv` to `/bin/iconv` and it now works.

Sample of the change 
![image](https://github.com/ZOSOpenTools/meta/assets/42328411/f1d14967-43c4-4624-8609-4530fb5b8355)

**Reply by lbdyck** [reply]
in this sample the `encoding` variable is `IBM-1047`

**Reply by IgorTodorovskiIBM** [reply]
It turns out that we're missing this patch in the stable release: https://github.com/ZOSOpenTools/libiconvport/commit/f798b59de885ac469b0d4f1f73f5463079bd2fbd

**Comment by lbdyck** [comment]
great catch

**Comment by fswarbrick** [comment]
Does the zopen version offer anything that the z/OS version does not?  Not sure why the zopen version even exists.
I thought it didn't even have EBCDIC support, as I tried using cp "1047".  Looks like you have to specify one of these instead: "CP1047 IBM-1047 IBM1047 CSIBM1047".

**Reply by IgorTodorovskiIBM** [reply]
Interesting, `/bin/iconv -l` yields 597 encodings while zopen iconv yields 197. @MikeFultonDev ported it so he may have a better idea as to what it offers.
